src/lib.rs

```diff
@@ -42,6 +43,32 @@ use rocksdb_store::{self as backend};
 // todo: Look into unifying these types?
 pub use backend::{GetError, SetError};
 
+enum StoreConstructorBundle<'a> {
+    Config(&'a StoreConfig),
```

Comment:
I'll rename `Config` to `DataDir`.

However, if you want `StoreConfig` renamed your gonna have to do that on your own. That's existing code you wrote that has nothing to do with my code. Not to mention you gave me 3 different options to choose from and I have no idea what you want and why you want it.

Reply:
Also, in my opinion `DataDirInit`, `AppCompany`, and `AppDefault` all sound bad as a replacement for `StoreConfig`.

That being said I can't really think of a better option either.

My advice would be to figure out what you want to rename it too and then make a change separate from this pull request. Because if you want to change `StoreConfig` to something else you will also need to change `fn new_from_config` in order for it to be consistent and that's not the goal of this pull request.

Reply:
I'm asking is because the name StoreConfig sounds like it should have all the possible configuration for a store, at least that's why I named it that.

As I see it, path is config, so it feels weird to have both a struct named config, and a config value that is not part of it.

Reply:
Fair enough.

Figure out what you want to rename `StoreConfig` to instead of giving me 3 options and I'll refactor it.

Reply:
`PlatformDefault` should work. And the new_from_config should now be new_in_location that all the other constructors forward to.

Also there may not be a reason for it it to be its own type anymore, if that's the case, just add the fields directly in the enum.

Reply:
> Also there may not be a reason for it it to be its own type anymore, if that's the case, just add the fields directly in the enum.

I see your point with this. For now I'm just going to refactor `new_from_config` and `StoreConfig` to `new_in_location` and `PlatformDefault` respectively.

I have a couple things I need to refocus my attention too after this pull request, so I'll most likely make a separate pull request replacing `PlatformDefault` with direct fields later down the line when I have more time.